src/Elastic.Apm/Model/Payload/Transaction.cs

```diff
@@ -33,6 +33,9 @@ public Transaction(AbstractLogger logger, string name, string type, IPayloadSend
 			Id = Guid.NewGuid();
 		}
 
+		/// <summary>
+		/// Any arbitrary contextual information regarding the event, captured by the agent, optionally provided by the user.
+		/// </summary>
 		public Context Context => _context.Value;
```

Comment:
internal as well,  even though `Transaction` is already external this clearly indicates its intend.  Similar with other properties/methods not on `ITransaction`.

Reply:
Also wondering why `_context`  and  `Context.tags` are `Lazy<T>`. The construction of either is not expensive in CPU nor memory.

Reply:
> internal as well, even though Transaction is already external this clearly indicates its intend. Similar with other properties/methods not on ITransaction.

This is an interesting comment. I spent time thinking about this and after reading [this](https://ericlippert.com/2014/09/15/internal-or-public/) I thought I'd use a similar approach. Now the question is: what do we mean by public fields on internal types?

> public: this member is not an implementation detail at all; it is the public surface

From what perspective is it public surface? And I thought that from the server's perspective it's public... it's public because we send it thorough the network and the server works with it, so it's public within the APM solution. What do you think about this? Of course I can make it internal and mark it for serialization, so nothing stops us from doing that.

> Also wondering why _context and Context.tags are Lazy<T>.
There are lots of scenarios where transactions and spans don't contains tags, so I thought not allocating anything is a good idea. Furthermore its also realistic that transactions or spans don't have a context, and it that case I think we still save a couple of allocation. ....we should measure this! :)

Reply:
@Mpdreamz I merged, but I think we can still discuss this (I'd be happy to hear what you think) and I'm ok with changing it.

Reply:
:thumbsup: on merge 

I'm happy with `public` coming around to it a second time. I have an irrational adversity against `internal` (and `sealed` but that's a separate discussion :smile:).  I actually never duplicate the `internal` modifier myself so not sure what triggered this comment on my part.